### PR TITLE
add more embed stuff

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

File: src/structures/RichEmbed.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+class RichEmbed {
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: Need a class JSDoc

File: src/structures/RichEmbed.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  constructor(data = {}) {
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: All properties need JSDocs

File: src/structures/RichEmbed.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+   * @returns {RichEmbed} This embed
+   */
+  setURL(url) {
+    this.url = url;
+    return this;
+  }
+
+  /**
+   * Sets the color of this embed
+   * @param {string|number|Array} color The color to set
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: Array of what?

File: src/structures/RichEmbed.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      color = parseInt(color.replace('#', ''), 16);
+    }
+    this.color = color;
+    return this;
+  }
+
+  /**
+   * Sets the author of this embed
+   * @param {string} name The name of the author
+   * @param {string} [icon=] The icon of the author
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: Don't need the `=`

File: src/structures/RichEmbed.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+   * @returns {RichEmbed} This embed
+   */
+  setAuthor(name, icon) {
+    this.author = { name, icon_url: icon };
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: `icon_url` needs to be camelCaseified. See the MessageEmbedAuthor class.

File: src/structures/RichEmbed.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    return this;
+  }
+
+  /**
+   * Sets the timestamp of this embed
+   * @param {Date} [timestamp=new Date()] The timestamp
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: Just say the default is `current time`

File: src/structures/RichEmbed.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+  /**
+   * Adds a field to the embed (max 25)
+   * @param {string} name The name of the field
+   * @param {string} value The value of the field
+   * @param {boolean} [inline=false] Set the field to display inline
+   * @returns {RichEmbed} This embed
+   */
+  addField(name, value, inline = false) {
+    if (this.fields.length >= 25) throw new Error('you may only add 25 fields to an embed!');
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: `A RichEmbed may only have a maximum of 25 fields.`
Also, RangeError.

File: src/structures/RichEmbed.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  /**
+   * Sets the footer of this embed
+   * @param {string} text The text of the footer
+   * @param {string} icon The icon of the footer
+   * @returns {RichEmbed} This embed
+   */
+  setFooter(text, icon) {
+    this.footer = { text, icon_uri: icon };
```
Comment (Gawdl3y) [inline src/structures/RichEmbed.js]: camelCase

File: src/structures/TextChannel.js [hunk 9]
```diff
@@ -74,6 +74,7 @@ class TextChannel extends GuildChannel {
 
   // These are here only for documentation purposes - they are implemented by TextBasedChannel
   sendMessage() { return; }
+  sendEmbed() { return; }
```
Comment (Gawdl3y) [inline src/structures/TextChannel.js]: Need this in User and GuildMember as well - also, sort it after `sendTTSMessage`

File: src/structures/interface/TextBasedChannel.js [hunk 10]
```diff
@@ -327,7 +349,7 @@ class TextBasedChannel {
 }
 
 exports.applyToClass = (structure, full = false) => {
-  const props = ['sendMessage', 'sendTTSMessage', 'sendFile', 'sendCode'];
+  const props = ['sendMessage', 'sendEmbed', 'sendTTSMessage', 'sendFile', 'sendCode'];
```
Comment (Gawdl3y) [inline src/structures/interface/TextBasedChannel.js]: Move `sendEmbed` to after `sendTTSMessage`

File: src/structures/interface/TextBasedChannel.js [hunk 11]
```diff
@@ -61,6 +62,27 @@ class TextBasedChannel {
   }
 
   /**
+   * Send an embed to this channel
+   * @param {Object|RichEmbed} embed The embed to send
+   * @param {Object|string} contentOrOptions Content to send or message options
+   * @param {Object} options If contentOrOptions is content, this will be options
+   * @returns {Promise<message>}
```
Comment (Gawdl3y) [inline src/structures/interface/TextBasedChannel.js]: Message needs capitalised

File: src/structures/interface/TextBasedChannel.js [hunk 12]
```diff
@@ -61,6 +62,27 @@ class TextBasedChannel {
   }
 
   /**
+   * Send an embed to this channel
+   * @param {Object|RichEmbed} embed The embed to send
```
Comment (Gawdl3y) [inline src/structures/interface/TextBasedChannel.js]: `RichEmbed` should be first

File: src/structures/interface/TextBasedChannel.js [hunk 13]
```diff
@@ -61,6 +62,27 @@ class TextBasedChannel {
...
+   * @param {Object|string} contentOrOptions Content to send or message options
```
Comment (Gawdl3y) [inline src/structures/interface/TextBasedChannel.js]: `string|MessageOptions`

File: src/structures/interface/TextBasedChannel.js [hunk 14]
```diff
@@ -61,6 +62,27 @@ class TextBasedChannel {
...
+   * @param {Object} options If contentOrOptions is content, this will be options
```
Comment (Gawdl3y) [inline src/structures/interface/TextBasedChannel.js]: `MessageOptions`, not `Object`

File: src/structures/RichEmbed.js (hunk 5 above)
Comment (GusCaplan) [inline src/structures/RichEmbed.js]: wrong. the property exists for discord, not for the end user. discord expects `icon_url`.

File: src/structures/RichEmbed.js (hunk 8 above)
Comment (GusCaplan) [inline src/structures/RichEmbed.js]: wrong. the property exists for discord, not for the end user. discord expects `icon_url`.

File: src/structures/RichEmbed.js (hunk 5 above)
Comment (GusCaplan) [inline src/structures/RichEmbed.js]: wrong. the property exists for discord, not for the end user. discord expects `icon_url`.

File: src/structures/RichEmbed.js (hunk 2 above)
Comment (GusCaplan) [inline src/structures/RichEmbed.js]: done

File: src/structures/RichEmbed.js (hunk 1 above)
Comment (GusCaplan) [inline src/structures/RichEmbed.js]: done

**Pull request merged**